### ISSUE TITLE
Anyone tried to reimplement DeepSequence in PyTorch?

### ISSUE THREAD

**Issue by charlesxu90** [debbiemarkslab/DeepSequence#11]
Dear authors,

I'm not sure if anyone tried to reimplement DeepSequence with PyTorch?

I wrote a version with the following link. 
https://github.com/charlesxu90/DeepSequence-torch

The result from the first batch is similar to the original version. But the later updates drastically changing the loss towards a very different direction, make it significantly lower than 0.

I check the parameters are correct. The first batch gives similar losses, so the equation implemented should be correct. This problem should be come from the gradient or the optimizer.

Any hints on this problem?

**Comment by charlesxu90** [comment]
Finally, I found out the reason. It comes from the loss.

The loss in the original implementation is a negative one. However, theano and the user-defined Adam seems able to process it as a positive value and minimize the absolute value of the loss.

However, if using PyTorch, the default gradient function and Adam optimizer continuously make it smaller, which make the absolute value of the loss even larger. 

I made the loss = - original_loss, and the problem solved.
